federatedscope/core/worker/server.py

```diff
@@ -276,13 +306,15 @@ def check_and_move_on(self,
         # round or finishing the evaluation
         if self.check_buffer(self.state, min_received_num, check_eval_result):
             if not check_eval_result:  # in the training process
```

Comment:
With the new PR, this "if" branch is too long (308~406), which is error-prone if we add more conditional branches within it. Plz try to simplify it into several sub-functions